pkg/virt-handler/vm.go

```diff
@@ -925,6 +925,27 @@ func (d *VirtualMachineController) execute(key string) error {
 	// Take different execution paths depending on the state of the migration and the
 	// node this is executed on.
 
+	if vmiExists && d.isMigrationSource(vmi) &&
+		vmi.Status.MigrationState.AbortRequested &&
+		!domain.Spec.Metadata.KubeVirt.Migration.Failed {
+		// Cancel the live migration if it's in progress but the migration
+		// object has been deleted
+		client, err := d.getLauncherClient(vmi)
+		if err != nil {
+			log.Log.Object(vmi).Reason(err).Errorf("failed to get virt-launcher client")
+			vmi.Status.MigrationState.AbortStatus = v1.MigrationAbortFailed
+			return fmt.Errorf("unable to create virt-launcher client connection: %v", err)
+		}
+		err = client.CancelVirtualMachineMigration(vmi)
+		if err != nil {
+			log.Log.Object(vmi).Reason(err).Errorf("failed to cancel live migration!")
+			vmi.Status.MigrationState.AbortStatus = v1.MigrationAbortFailed
+			return err
+		}
+		vmi.Status.MigrationState.AbortStatus = v1.MigrationAbortSucceeded
```

Comment:
That needs to go to the status update function. You will have to observe in a subsequent loop run from virt-launcher that the migration was aborted. Otherwise it messes up the controller architecture and may have strange side-effects, like the libvirt call succeeded, but our domain object still reports that the migration is in progress (race-conditions) and you may see status flips.